test/integration/index-test.js

```diff
@@ -250,6 +253,62 @@ describe(function() {
         assertNoUnstaged(status);
       });
 
+      it('can update a custom blueprint for an ember app project', async function() {
+        let originalSpawn = installAndGenerateBlueprint.spawn;
+        let originalResolvePackage = cliUpdateCommandModule.resolvePackage;
+        let originalIsDefaultAddonBlueprint = getStartAndEndModule.isDefaultAddonBlueprint;
+        let fakeAddonName = 'some-addon-with-blueprint';
+
+        sinon.stub(getStartAndEndModule, 'isDefaultAddonBlueprint').callsFake((blueprint) => {
+          if (blueprint.packageName === fakeAddonName) {
+            return true;
+          }
+          return originalIsDefaultAddonBlueprint(blueprint);
+        });
+        // Mock this for the fake addon
+        sinon.stub(cliUpdateCommandModule, 'resolvePackage').callsFake(({ name, url, range }) => {
+          if (name  === fakeAddonName) {
+            return {
+              version: range,
+              path: ''
+            };
+          } else {
+            return originalResolvePackage({ name, url, range });
+          }
+        });
+        // Mock out the spawn to install a local blueprint
+        sinon.stub(installAndGenerateBlueprint, 'spawn').callsFake((bin, args, options) => {
+          if (args.some(arg => {
+            return arg.indexOf(fakeAddonName) > -1;
+          })
+          ) {
+            let [installCmd, saveDev, resolvedPackageName] = args;
+            let packageNameAndVersion = resolvedPackageName.split('@');
+            let absolutePath = path.resolve(path.join('test/fixtures/blueprint/addon/legacy', `v${packageNameAndVersion[1]}`));
+            return originalSpawn(bin, [installCmd, saveDev, absolutePath], options);
+          } else {
+            return originalSpawn(bin, args, options);
+          }
+        });
+
+        let {
+          status
+        } = await merge({
+          fixturesPath: 'test/fixtures/app/simple-app-custom-blueprint-test',
```

Comment:
you'll want the "test/fixtures/blueprint/app" folder which is where the custom app blueprints are for testing.

Reply:
I may need clarity on what you mean.

The use case I had in mind for the new functionality was that an addon houses blueprints to bootstrap. When I think of `custom` blueprint i think of the non default blueprint for an addon.

I can copy over an addon fixture to the `blueprint/app` folder but the functionality I'm testing is having the app upgrade the version addon containing the blueprint to be applied

Reply:
fixtures/app is testing the standard ember-cli blueprint. You'll want something like test/fixtures/blueprint/app/non-default-addon-blueprint-app/{local,merge}

Reply:
Moved `simple-app-custom-blueprint-test` and `simple-app-custom-blueprint-test-merge` to `fixtures/blueprint/app/non-default-addon-blueprint-app/{local,merge}`